haystack/nodes/retriever/link_content.py

```diff
@@ -177,24 +220,56 @@ def run_batch(
 
         return {"documents": results}, "output_1"
 
-    def _get_response(self, url: str, timeout: Optional[int]) -> requests.Response:
+    def _get_response(self, url: str, timeout: Optional[int] = None) -> requests.Response:
         """
         Fetches content from a URL. Returns a response object.
         :param url: The URL to fetch content from.
         :param timeout: The timeout in seconds.
         :return: A response object.
         """
+
+        @retry(
+            reraise=True,
+            stop=stop_after_attempt(self.retry_attempts),
+            wait=wait_exponential(multiplier=1, min=2, max=10),
+            retry=(retry_if_exception_type((HTTPError, requests.RequestException))),
+            after=self._switch_user_agent,
+        )
+        def _request():
+            headers = self.REQUEST_HEADERS.copy()
+            headers["User-Agent"] = self.current_user_agent
+            r = requests.get(url, headers=headers, timeout=timeout or 3)
+            r.raise_for_status()
+            return r
+
         try:
-            response = requests.get(url, headers=LinkContentFetcher.REQUEST_HEADERS, timeout=timeout)
-            response.raise_for_status()
+            response = _request()
         except Exception as e:
             if self.raise_on_failure:
                 raise e
 
             logger.warning("Couldn't retrieve content from %s", url)
             response = requests.Response()
+        finally:
+            self.current_user_agent = self.default_user_agent
         return response
 
+    def _get_content_type_handler(self, content_type: str) -> Callable:
+        """
+        Get the appropriate content handler based on the content type.
+        :param content_type: The content type of the response.
+        :return: The matching content handler callable or the default html_content_handler if no match is found.
+        """
+        mime_type: str = (content_type or "").split(";")[0]
+        return self.handlers[mime_type]
+
+    def _switch_user_agent(self, retry_state: RetryCallState) -> None:
+        """
+        Switches the User-Agent for this LinkContentRetriever to the next one in the list of user agents.
+        :param retry_state: The retry state (unused, required by tenacity)
+        """
+        self.current_user_agent = next(self.user_agents)
+
     def _is_valid_url(self, url: str) -> bool:
         """
```

Comment:
**Suggestion:** Consider using a more robust URL validation method. The current method `_is_valid_url` only checks if the URL is not None or empty, which is not sufficient to ensure that the URL is valid.
```suggestion
    from urllib.parse import urlparse

    def _is_valid_url(self, url: str) -> bool:
        try:
            result = urlparse(url)
            return all([result.scheme, result.netloc])
        except ValueError:
            return False
```